Docker_cram/dockerfile

```diff
@@ -2,103 +2,60 @@
 FROM osrf/ros:noetic-desktop-full
 SHELL ["/bin/bash", "-c"]
 
-#source ROS
-RUN source /opt/ros/noetic/setup.bash
-
-#Debug
-RUN echo $CMAKE_PREFIX_PATH
-RUN echo $ROS_PACKAGE_PATH
+ENV ROSINSTALL_PREFIX=https://raw.githubusercontent.com/artnie/cram/jupyter-noetic-devel
+ENV CRAM_ROSINSTALL_FILE="${ROSINSTALL_PREFIX}/jupyter/cram-20.04-http.rosinstall"
+ENV JUPYTER_ROSINSTALL_FILE="${ROSINSTALL_PREFIX}/jupyter/common-lisp-jupyter.rosinstall"
 
-#make directory
-RUN mkdir -p workspace/ros
-
-#Explicitly source ros
+# ------
+RUN echo "--- Set up ROS environment variables ---"
+RUN source /opt/ros/noetic/setup.bash
 ENV ROS_PACKAGE_PATH=/home/workspace/ros/src:/opt/ros/noetic/share:/opt/ros/noetic/stacks
 ENV CMAKE_PREFIX_PATH=/home/workspace/ros/devel:/opt/ros/noetic
-
-#check ROS Sourcing for debugging
-RUN echo $ROS_PACKAGE_PATH
-RUN echo $CMAKE_PREFIX_PATH
-
-# #install Juypterlab
+ENV PATH=/opt/ros/noetic/bin:$PATH
+ENV ROS_ROOT=/opt/ros/noetic/share/ros
 ENV PYTHONPATH=/home/workspace/ros/devel/lib/python3/dist-packages:/opt/ros/noetic/lib/python3/dist-packages
-RUN sudo apt-get update && apt-get upgrade -y 
-RUN sudo apt-get install -y python3-pip python3-dev
-RUN sudo -H pip3 install --upgrade pip
-RUN sudo -H pip3 install jupyterlab
-
-#installing python3 catkin tools and dependencies
-RUN sudo apt-get update && apt-get upgrade -y 
-RUN sudo apt-get install -y python3-catkin-tools
-RUN sudo apt-get install -y apt-utils
-RUN sudo apt-get install -y python3-rosdep python3-rosinstall python3-rosinstall-generator python3-wstool build-essential
-RUN sudo apt-get install -y ros-noetic-roslisp-repl
-
-#setting up workspace for CRAM
-RUN sudo apt-get update && apt-get upgrade -y 
-RUN apt-get install -y ros-noetic-roslisp-repl
-RUN apt-get install -y python3-rosinstall python3-wstool
+ENV PATH=/root/.local/bin:$PATH
+RUN rosdep update # using as root is not intended, it works though.
+
+# ------
+RUN echo "--- Install ROS tools and dependencies ---"
+RUN apt-get update && apt-get upgrade -y 
+RUN apt-get install -y dialog curl git python3-dev python3-pip python3-catkin-tools python3-rosdep \
+python3-rosinstall python3-rosinstall-generator python3-wstool build-essential ros-noetic-roslisp-repl
+
+# ------
+RUN echo "--- Set up Catkin Workspace ---"
+RUN mkdir -p /home/workspace/ros/src 
 WORKDIR /home/workspace/ros
-RUN echo $PWD
 RUN catkin init
-WORKDIR src
-RUN echo $PWD
+WORKDIR /home/workspace/ros/src
```

Comment:
we can only write WORKDIR src, WORKDIR acts as cd, but this is also fine

Reply:
again, the red lines are removed and the green ones are new. i figured that WORKDIR is even more powerful and dows `mkdir -p <path>` and `cd <path>` simultaneously, which is very nice